src/Operators/remapping.jl

```diff
@@ -103,6 +93,66 @@ function overlap_weights(
     return kron(Y_ov, X_ov)
 end
 
+function overlap_weights(
+    target::T,
+    source::S,
+) where {
+    T <: SpectralElementSpace1D{<:IntervalTopology},
+    S <: SpectralElementSpace1D{<:IntervalTopology, Quadratures.GL{1}},
+}
+    FT = Spaces.undertype(source)
+    target_topo = Spaces.topology(target)
+    source_topo = Spaces.topology(source)
+    nelems_t = Topologies.nlocalelems(target)
+    nelems_s = Topologies.nlocalelems(source)
+    QS_t = Spaces.quadrature_style(target)
+    QS_s = Spaces.quadrature_style(source)
+    Nq_t = Quadratures.degrees_of_freedom(QS_t)
+    Nq_s = Quadratures.degrees_of_freedom(QS_s)
+    J_ov = spzeros(nelems_t * Nq_t, nelems_s * Nq_s)
+
+    # Calculate element overlap pattern
+    # X_ov[i,j] = overlap length between target elem i and source elem j
+    for i in 1:nelems_t
+        vertices_i = Topologies.vertex_coordinates(target_topo, i)
+        min_i, max_i = Geometry.component(vertices_i[1], 1),
+        Geometry.component(vertices_i[2], 1)
+        for j in 1:nelems_s # elems coincide w nodes in FV source
+            vertices_j = Topologies.vertex_coordinates(source_topo, j)
+            # get interval for quadrature
+            min_j, max_j = Geometry.component(vertices_j[1], 1),
+            Geometry.component(vertices_j[2], 1)
+            min_ov, max_ov = max(min_i, min_j), min(max_i, max_j)
+            if max_ov <= min_ov
+                continue
+            end
+            ξ, w = Quadratures.quadrature_points(FT, QS_t)
+            x_ov =
+                FT(0.5) * (min_ov + max_ov) .+ FT(0.5) * (max_ov - min_ov) * ξ
+            x_t = FT(0.5) * (min_i + max_i) .+ FT(0.5) * (max_i - min_i) * ξ
+
+            I_mat = Quadratures.interpolation_matrix(x_ov, x_t)
+            for k in 1:Nq_t
+                idx = Nq_t * (i - 1) + k
+                # (integral of basis on overlap) / (reference elem length * overlap elem length)
+                J_ov[idx, j] = w' * I_mat[:, k] ./ 2 * (max_ov - min_ov)
```

Comment:
Thanks Ben for your valuable contribution!
I would specify in the comment that the k-th column of the interpolant matrix gives you the k-th basis function defined on the element